src/Workspaces/Core/Portable/Shared/Extensions/ISolutionExtensions.cs

```diff
@@ -66,16 +66,7 @@ public static Solution WithTextDocumentText(this Solution solution, DocumentId d
             }
         }
 
-        public static IEnumerable<DocumentId> FilterDocumentIdsByLanguage(this Solution solution, ImmutableArray<DocumentId> documentIds, string language)
-        {
-            foreach (var documentId in documentIds)
-            {
-                var document = solution.GetDocument(documentId);
-                if (document != null && document.Project.Language == language)
-                {
-                    yield return documentId;
-                }
-            }
-        }
+        public static ImmutableArray<DocumentId> FilterDocumentIdsByLanguage(this Solution solution, ImmutableArray<DocumentId> documentIds, string language)
+            => documentIds.WhereAsArray((documentId, language) => solution.GetDocument(documentId)?.Project.Language == language, language));
```

Comment:
>solution [](start = 66, length = 8)

`solution` is captured.

Reply:
Ah, I didn't mean to include the IDE changes... removing.